test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1

```diff
@@ -103,6 +108,61 @@ Describe "Basic FileSystem Provider Tests" -Tags "CI" {
             $contentBefore.Count | Should Be 1
             $contentAfter.Count | Should Be 0
         }
+
+         It "Copy-Item rejects Windows reserved device names" -Skip:(-not $IsWindows) {
+             foreach ($deviceName in $reservedNames)
+             {
+                { Copy-Item -Path $testFile -Destination $deviceName -ErrorAction Stop } | ShouldBeErrorId "CopyError,Microsoft.PowerShell.Commands.CopyItemCommand"
+             }
+         }
+
+         It "Move-Item rejects Windows reserved device names" -Skip:(-not $IsWindows) {
+             foreach ($deviceName in $reservedNames)
+             {
+                { Move-Item -Path $testFile -Destination $deviceName -ErrorAction Stop } | ShouldBeErrorId "MoveError,Microsoft.PowerShell.Commands.MoveItemCommand"
+             }
+         }
+
+         It "Rename-Item rejects Windows reserved device names" -Skip:(-not $IsWindows) {
+             foreach ($deviceName in $reservedNames)
+             {
+                { Rename-Item -Path $testFile -NewName $deviceName -ErrorAction Stop } | ShouldBeErrorId "RenameError,Microsoft.PowerShell.Commands.RenameItemCommand"
+             }
+         }
+
+         It "Copy-Item succeeds with Windows reserved device names" -Skip:($IsWindows) {
+             foreach ($deviceName in $reservedNames)
+             {
+                $newFile = Copy-Item -Path $testFile -Destination $deviceName -PassThru
+                $fileExists = Test-Path $deviceName
+                $fileExists | Should Be $true
+                $newFile.Name | Should Be $deviceName
+             }
+         }
+
+         It "Move-Item succeeds with Windows reserved device names" -Skip:($IsWindows) {
+             foreach ($deviceName in $reservedNames)
+             {
+                Copy-Item -Path $testFile -Destination $tempFile -Force
```

Comment:
For Move-Item and Rename-Item the $tempFile was to be able to restore $testFile each time through the foreach across device names. I can certainly use New-Item to create a new $testFile each iteration if you prefer.